src/librustc/diagnostics.rs

```diff
@@ -151,6 +191,37 @@ loop {
 }
 "##,
 
+E0170: r##"
+Enum variants are qualified by default. For example, given this type:
+
+enum Method {
+    GET,
+    POST
+}
+
+you would match it using:
+
+match m {
+    Method::GET => ...
+    Method::POST => ...
+}
+
+If you don't qualify the names, the code will bind new variables named "GET" and
+"POST" instead. This behavior is likely not what you want, so rustc warns when
+that happens.
+
+Qualified names are good practice, and most code works well with them. But if
+you prefer them unqualified, you can import the variants into scope:
+
+// different module
+use http::Method;
+use http::Method::*;
```

Comment:
This reference to `http` may be a little confusing as it's not mentioned anywhere else, perhaps just the glob import below could suffice?